src/plugins/integrations/RedisIntegration.ts

```diff
@@ -40,66 +42,71 @@ class RedisIntegration implements Integration {
-            [SpanTags.TRIGGER_OPERATION_NAMES]: [tracer.functionName],
-          },
-        });
-
-        const originalCallback = options.callback;
-
-        const wrappedCallback = (err: any, res: any) => {
-          if (err) {
-            const parseError = Utils.parseError(err);
-            span.setTag('error', true);
-            span.setTag('error.kind', parseError.errorType);
-            span.setTag('error.message', parseError.errorMessage);
+          if (!options) {
+            return internalSendCommand.call(this, options);
           }
 
-          span.close();
+          const parentSpan = tracer.getActiveSpan();
+          let host = 'localhost';
+          let port = '6379';
+          let command = '';
 
-          originalCallback(err, res);
-        };
+          if (this.connection_options) {
+            host = String(this.connection_options.host);
+            port = String(this.connection_options.port);
+            command = options.command.toUpperCase();
+          }
 
-        options.callback = wrappedCallback;
+          const operationType = RedisCommandTypes[command] ? RedisCommandTypes[command] : 'READ';
+
+          const span = tracer._startSpan(host, {
+            childOf: parentSpan,
+            domainName: DomainNames.CACHE,
+            className: ClassNames.REDIS,
+            disableActiveStart: true,
+            tags: {
+              [SpanTags.SPAN_TYPE]: SpanTypes.REDIS,
+              [DB_TYPE]: DBTypes.REDIS,
+              [DB_INSTANCE]: host,
+              [DBTags.DB_STATEMENT_TYPE]: operationType,
+              [RedisTags.REDIS_HOST]: host,
+              [RedisTags.REDIS_PORT]: port,
+              [RedisTags.REDIS_COMMAND]: command,
+              [RedisTags.REDIS_COMMAND_TYPE]: operationType,
+              [RedisTags.REDIS_COMMAND_ARGS]: options.args.join(','),
+              [SpanTags.OPERATION_TYPE]: operationType,
+              [SpanTags.TOPOLOGY_VERTEX]: true,
+              [SpanTags.TRIGGER_DOMAIN_NAME]: LAMBDA_APPLICATION_DOMAIN_NAME,
+              [SpanTags.TRIGGER_CLASS_NAME]: LAMBDA_APPLICATION_CLASS_NAME,
+              [SpanTags.TRIGGER_OPERATION_NAMES]: [tracer.functionName],
+            },
+          });
+
+          const originalCallback = options.callback;
+
+          const wrappedCallback = (err: any, res: any) => {
+            if (err) {
+              const parseError = Utils.parseError(err);
+              span.setTag('error', true);
+              span.setTag('error.kind', parseError.errorType);
+              span.setTag('error.message', parseError.errorMessage);
+            }
+
+            span.close();
+
+            originalCallback(err, res);
+          };
+
+          options.callback = wrappedCallback;
 
-        return internalSendCommand.call(this, options);
+          return internalSendCommand.call(this, options);
+        } catch (error) {
+          ThundraLogger.getInstance().error(error);
+          internalSendCommand.call(this, options);
```

Comment:
In here, if `internalSendCommand` command itself fails, do we send another call to Redis?

Reply:
In case of error in the wrapped function error callback is called. I verified, it is not called twice.